contexts/SupabaseContext.tsx

```diff
@@ -17,43 +39,89 @@ interface SupabaseContextInterface {
       error: ApiError | null
     }>
   );
-  signUp: ({ email, password, phone }: UserCredentials) => Promise<{
+  signUp: ({ email, password }: UserCredentials, options: { redirectTo: string }) => Promise<{
     user: User | null
     session: Session | null
     error: ApiError | null
   }>;
   signOut: () => Promise<{ error: ApiError | null }>;
-  user: User;
+  // TODO: Would be nice if these weren't nullable under `ProtectedRoute`s
+  user: EmailUser | GoogleUser | null;
+  onboardingStep: OnboardingStep | null;
```

Comment:
If the current component is a child of `ProtectedRoute`, we know that `user` and `onboardingStep` can't be `null`, but I haven't come up with a good way to avoid checking this in every single protected route :grin: Might require a bit of a redesign...